scripts/python/verify_agent_os.py

```diff
@@ -1,14 +1,25 @@
 # Script verifies the operating system for the platform on which it is being run
 # Used in build pipelines to verify the build agent os
-# Variable: The name of the os to verfy against
-
+# Variable: The friendly name or image name of the os to verfy against
+from __future__ import print_function
 import sys
 import platform
-os_parameter = sys.argv[1]
-print ("Parameter passed by matrix -", os_parameter)
-os_parameter = 'Darwin' if(os_parameter.lower() == 'MacOS'.lower()) else os_parameter
+
+os_parameter = sys.argv[1].lower()
+if os_parameter.startswith('mac') or os_parameter.startswith('darwin'):
+    os_parameter = 'macOS'
+elif os_parameter.startswith('ubuntu') or os_parameter.startswith('linux'):
+    os_parameter = 'Linux'
+elif os_parameter.startswith('vs') or os_parameter.startswith('win'):
+    os_parameter = 'Windows'
+
+print("Job requested to run on OS: %s" % (os_parameter))
+
 agent_os = platform.system()
+agent_os = 'macOS' if agent_os == 'Darwin' else agent_os
+
 if (agent_os.lower() == os_parameter.lower()):
-	print ('Job ran on %s OS' %agent_os)
+    print('Job ran on OS: %s' % (agent_os))
+    print('##vso[task.setvariable variable=OSName]%s' % (agent_os))
```

Comment:
@bsiegel Can you remember why you added this line? The OSName variable is intended to be readonly and overwriting it creates a warning in out pipelines. Soon overwriting will not be possible.  Example [here](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=275541&view=logs&j=24e84196-b4b8-5123-c7ea-5e853df7a17a&t=4d9fffad-7196-5e77-10e3-02e9c7c4dee0).

I am looking to update this script to resolve the issue but I have been unable to determine why the variable is being set here in the first place.

Reply:
No, but I would guess that at the time, this was the only way we were setting the `OSName` variable. If we're setting it elsewhere in the pipeline now, it may not have been that way back when this PR was merged.

Reply:
Ok, I'll remove it and see if things continue working as intended.